core/amber/src/main/scala/edu/uci/ics/amber/engine/faulttolerance/scanner/BufferedBlockReader.java

```diff
@@ -48,38 +48,14 @@ public BufferedBlockReader(InputStream input, long blockSize, char delimiter, in
             int start = cursor;
             while (cursor < bufferSize) {
                 if (buffer[cursor] == delimiter) {
-                    if(keptFields == null || keptFields.contains(index)){
-                        if(cursor-start > 0){
-                            outputStream.write(buffer,start,cursor-start);
-                            fields.add(outputStream.toString());
-                        }else if(outputStream.size()>0){
-                            fields.add(outputStream.toString());
-                        }else{
-                            fields.add(null);
-                        }
-                    }
+                    addField(start,index);
                     outputStream.reset();
-                    currentPos += cursor - start + 1;
                     start = cursor+1;
                     index++;
-                }else if(buffer[cursor] == '\n'){
-                    if(keptFields == null || keptFields.contains(index)){
-                        if(cursor-start > 0){
-                            if(buffer[cursor-1] != '\r') {
-                                outputStream.write(buffer,start,cursor-start);
-                            }else{
-                                outputStream.write(buffer,start,cursor-start-1);
-                            }
-                            fields.add(outputStream.toString());
-                        }else if(outputStream.size()>0){
-                            fields.add(outputStream.toString());
-                        }else{
-                            fields.add(null);
-                        }
-                    }
-                    currentPos += cursor - start + 1;
+                }else if(buffer[cursor] == '\r' || buffer[cursor] == '\n'){
+                    addField(start,index);
```

Comment:
I am having a very hard time understanding the logic of this piece of code, especially of the case of `\r\n`, and I'm also trying to understand what will happens in case of `\r\n`, but `\r` is the last byte of a buffer and `\n` is the first byte of new buffer. Also what's the meaning of adding `null` to the field? 

I think this piece of code has a potential of doing a bit refactor and make the logic cleaner, or at least add some comments to explain the logic? That would be good.

Reply:
added some comments to explain the behavior.